moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -532,11 +534,40 @@ void PlanningScene::setCollisionObjectUpdateCallback(const collision_detection::
   current_world_object_update_callback_ = callback;
 }
 
-collision_detection::AllowedCollisionMatrix& PlanningScene::getAllowedCollisionMatrixNonConst()
+void PlanningScene::update(collision_detection::AllowedCollisionMatrix const& new_acm)
 {
-  if (!acm_)
-    acm_ = std::make_shared<collision_detection::AllowedCollisionMatrix>(parent_->getAllowedCollisionMatrix());
-  return *acm_;
+  acm_ = std::make_shared<collision_detection::AllowedCollisionMatrix>(new_acm);
```

Comment:
If internally, this is stored as a shared_ptr, maybe we should make it a shared_ptr to const, and the update parameter should take a shared_ptr to const. The way you wrote it here it takes the acm as a parameter on the stack and then allocates a acm on the heap inside this shared ptr and copies it into the shared_ptr. While this is safe, it is creating a bunch of unnecessary copies.